ultravox/tools/infer_tool.py

```diff
@@ -190,25 +192,48 @@ def dataset_infer(inference: base.VoiceInference, args: InferArgs):
     if args.seed is not None:
         ds_args.shuffle_seed = args.seed
     ds = datasets.create_dataset(args.data_sets[0], ds_args)
-    scores: List[float] = []
-    for i, sample in enumerate(datasets.Range(ds, args.num_samples)):
-        # Store the original question and answer for JSON output.
-        question_text = sample.audio_transcript
-        expected_answer = sample.messages[-1]["content"]
-        # Drop any assistant response from the sample.
-        sample.messages = sample.messages[:-1]
-        if not args.json:
-            run_tui(i, inference, sample, args, expected_answer, scores)
-        else:
-            output = inference.infer(
-                sample, max_tokens=args.max_tokens, temperature=args.temperature
+
+    if args.json and isinstance(inference, infer.LocalInference):
```

Comment:
I tend to think we should degrade gracefully here and just simulate `infer_batch` when an optimized batch API doesn't exist, via looping. The actual looping implementation could be in the base class.

Reply:
Otherwise we'll malfunction if args.json is set and a different impl is used.